knowledge_verificator/answer_chooser.py

```diff
@@ -12,6 +13,7 @@ class AnswerChooser:
     """
 
     def __init__(self) -> None:
+        self._cache: dict[str, str | None] = {}
```

Comment:
I would say that cache should contain not last chosen anwser, but list of all useful PoS for given paragraph

Reply:
Okay, I have changed `self._cache` store all possible answers.